### PR TITLE
fix(grafana-apps): copy namespace from connect ctx

### PR DESCRIPTION
**Pull request by JohnnyQQQQ**
This pull request addresses the issue of a missing namespace within the HTTP request context. Previously, the namespace was exclusively provided by the Connect context, limiting its availability downstream. This update ensures the namespace is propagated into the HTTP request context, thereby making it accessible to subsequent functions and enhancing overall context propagation throughout the call chain.

### CODE REVIEW

File: pkg/registry/apis/query/query.go [hunk 1]
```diff
@@ -86,7 +87,8 @@ func (r *queryREST) Connect(ctx context.Context, name string, opts runtime.Objec
 	b := r.builder
 
 	return http.HandlerFunc(func(w http.ResponseWriter, httpreq *http.Request) {
-		ctx, span := b.tracer.Start(httpreq.Context(), "QueryService.Query")
+		ctx := request.WithNamespace(httpreq.Context(), request.NamespaceValue(connectCtx))
+		ctx, span := b.tracer.Start(ctx, "QueryService.Query")
```
Comment (toddtreece) [inline pkg/registry/apis/query/query.go:91]: would it be useful to also add it to the http req context? 

```suggestion
		ctx, span := b.tracer.Start(ctx, "QueryService.Query")
		httpreq = httpreq.WithContext(ctx)
```

**Review by toddtreece — APPROVED** [review]

File: pkg/registry/apis/query/query.go (hunk 1 above)
Comment (JohnnyQQQQ) [reply pkg/registry/apis/query/query.go:91]: Yeah, it should probably be the other way round.

Comment (toddtreece) [reply pkg/registry/apis/query/query.go]: > it should probably be the other way round

do you mean something like this?
```go
httpreq = httpreq.WithContext(connectCtx)
```

Comment (JohnnyQQQQ) [reply pkg/registry/apis/query/query.go]: No, just that I add first the tracer to the httpreq, and then create the new context. I don't think that it has any implications, as the span is closed after the function anyway, but it looks and feels cleaner.

Comment (toddtreece) [reply pkg/registry/apis/query/query.go]: is the namespace only needed by `parseRequest`?

Comment (JohnnyQQQQ) [reply pkg/registry/apis/query/query.go]: For now yes, it's only needed by `parseRequest`.

**Review by toddtreece — APPROVED** [review]

**Pull request merged**